fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -594,14 +597,15 @@ protected void replaceResourceBinaryWithStream(final FedoraBinary result,
                                                    final InputStream requestBodyStream,
                                                    final ContentDisposition contentDisposition,
                                                    final MediaType contentType,
-                                                   final String checksum) throws InvalidChecksumException {
-        final URI checksumURI = checksumURI(checksum);
+                                                   final Collection<String> checksums) throws InvalidChecksumException {
+        final Collection<URI> checksumURIs = null == checksums ?
+                new HashSet<>() : checksums.stream().map(checksum -> checksumURI(checksum)).collect(Collectors.toSet());
```

Comment:
* `null == checksums` prevents the accidental `checksums = null`... but I am happy to please.

* Initially I had a method reference, but have decided to keep it as-is since the method reference is actually longer in this case: `ContentExposingResource::checksumURI`.